uCNC/src/core/motion_control.c

```diff
@@ -1202,14 +1202,13 @@ uint8_t mc_build_hmap(float *target, float *offset, float retract_h, motion_data
 
 	block_data->feed = FLT_MAX;
 	// fast retract if needed
-	if (position[AXIS_TOOL] < (target[AXIS_TOOL] + retract_h))
+	// retract (higher if needed)
+	block_data->feed = FLT_MAX;
+	position[AXIS_TOOL] = MAX((target[AXIS_TOOL] + retract_h), minretract_h);
```

Comment:
not that it makes much difference, but using `position` here would be more consistent with the calculation for the retraction at the probe points (as `target` is below the workpiece).

Reply:
As I said. This forces the retraction to be **at least** the initial starting probe position + R and not the previous probe + R.

This ensures it will rise to a safe position although it might be longer then R in some cases

Reply:
Right, but here target is below the workpiece, whereas position is not. Both are ok.